platform/linux-generic/pktio/socket.c

```diff
@@ -855,6 +937,115 @@ static int sock_init_global(void)
 	return 0;
 }
 
+static int sock_recv_mq_tmo_select(pktio_entry_t * const *entry,
+				   const int index[],
+				   unsigned num_q, unsigned *from,
+				   odp_packet_t packets[], int num,
+				   uint64_t usecs, fd_set *readfds,
+				   int maxfd)
+{
+	struct timeval timeout;
+	unsigned i;
+	int ret;
+
+	timeout.tv_sec = usecs / 1000 / 1000;
+	timeout.tv_usec = usecs % (1000 * 1000);
+
+	for (i = 0; i < num_q; i++) {
+		ret = entry[i]->s.ops->recv(entry[i], index[i], packets, num);
+
+		if (ret > 0 && from)
+			*from = i;
+
+		if (ret != 0)
+			return ret;
+	}
+
+	if (select(maxfd + 1, readfds, NULL, NULL,
+		   usecs == ODP_PKTIN_WAIT ? NULL : &timeout) == 0)
+		return 0;
+
+	for (i = 0; i < num_q; i++) {
+		ret = entry[i]->s.ops->recv(entry[i], index[i], packets, num);
+
+		if (ret > 0 && from)
+			*from = i;
+
+		if (ret != 0)
+			return ret;
+	}
+
+	return 0;
+}
+
+int sock_recv_mq_tmo_try_int_driven(const struct odp_pktin_queue_t queues[],
```

Comment:
Couple comment lines describing the different stages of this function wouldn't hurt when someone is reading this code later on.